### PR TITLE
DOCSP-26447: use struct in write ops

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-26447
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-26447-struct-write-ops/fundamentals/crud/write-operations/#std-label-golang-crud-write-operations

**Pages changed:**
1. bulk operations
2. insert a document
3. delete a document
4. update arrays
5. insert or update

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?

### CODE REVIEW

File: source/fundamentals/crud/write-operations/embedded-arrays.txt [hunk 1]
```diff
@@ -15,7 +15,8 @@ Update Arrays in a Document
 Overview
 --------
 
-In this guide, you can learn how to update array elements in a document.
+In this guide, you can learn how to update array elements in a document
+or multiple documents.
```
Comment (jordan-smith721) [inline source/fundamentals/crud/write-operations/embedded-arrays.txt:19]: Minor suggestion (feel free to take or leave this one) but "a document or multiple documents" feels too wordy to me, maybe something like:
```suggestion
In this guide, you can learn how to update array elements in one,
or multiple documents.
```

File: source/fundamentals/crud/write-operations/embedded-arrays.txt [hunk 2]
```diff
@@ -26,15 +27,31 @@ To update elements in an array, perform the following actions:
 Sample Data
 ~~~~~~~~~~~
 
+The examples in this guide use the following ``Book`` struct as a model for documents
+in the ``books`` collection:
```
Comment (jordan-smith721) [inline source/fundamentals/crud/write-operations/embedded-arrays.txt:31]:
```suggestion
The examples in this guide use the following ``Drink`` struct as a model for documents
in the ``drinks`` collection:
```

File: source/fundamentals/crud/write-operations/embedded-arrays.txt [hunk 3]
```diff
@@ -26,15 +27,31 @@ To update elements in an array, perform the following actions:
...
 .. literalinclude:: /includes/fundamentals/code-snippets/CRUD/updateArray.go
    :language: go
    :dedent:
-   :start-after: begin insert docs
-   :end-before: end insert docs
+   :start-after: begin insertDocs
+   :end-before: end insertDocs
+
+Each document contains a description of a drink that
+includes the drink's description, available sizes (in ounces), and available
```
Comment (jordan-smith721) [inline source/fundamentals/crud/write-operations/embedded-arrays.txt:52]: Suggestion: I would try to avoid parenthesis if possible

```suggestion
includes the drink's description, available ounce sizes, and available
```

or

```suggestion
includes the drink's description, available sizes in ounces, and available
```

File: source/fundamentals/crud/write-operations/insert.txt [hunk 4]
```diff
@@ -206,10 +212,10 @@ Assume you want to insert the following documents:
...
    :copyable: false
 
-   { "_id": 1, "country": "Tanzania" }
-   { "_id": 2, "country": "Lithuania" }
-   { "_id": 1, "country": "Vietnam" }
-   { "_id": 3, "country": "Argentina" }
+   { "_id": 1, "name": "Tanzania" }
+   { "_id": 2, "name": "Lithuania" }
+   { "_id": 1, "name": "Vietnam" }
+   { "_id": 3, "name": "Argentina" }
```
Comment (jordan-smith721) [inline source/fundamentals/crud/write-operations/insert.txt:218]: Suggestion: It feels a bit jarring to me reading through this to suddenly be talking about locations rather than books, since the rest of the page is using the books collection. Could we maybe adjust this a bit so that it is using something related to books? I don't think it necessarily has to fit into the struct exactly, but it could at least be in the same subject matter.

**Review by jordan-smith721 — CHANGES_REQUESTED** [review]
Nice work! Just a couple minor suggestions and changes

File: source/fundamentals/crud/write-operations/embedded-arrays.txt (hunk 1 above)
Comment (rustagir) [reply source/fundamentals/crud/write-operations/embedded-arrays.txt:19]: decided to go with "one or more documents"

File: source/fundamentals/crud/write-operations/embedded-arrays.txt (hunk 2 above)
Comment (rustagir) [reply source/fundamentals/crud/write-operations/embedded-arrays.txt:31]: good catch!

File: source/fundamentals/crud/write-operations/insert.txt (hunk 4 above)
Comment (rustagir) [reply source/fundamentals/crud/write-operations/insert.txt:218]: changed to `Book` again with less fields for brevity.

**Review by jordan-smith721 — APPROVED** [review]
LGTM!
